### PR TITLE
[xpu] add dropout & amp ops in xpu place

### PR DESCRIPTION
**Pull request by taixiurong**
### PR types
Others

### PR changes
OPs

### Describe
1. add new xpu api of dropout 2. add xpu amp ops 3. fix elewise_add crash when dy is null in xpu place
test=kunlun

### CODE REVIEW

File: paddle/fluid/operators/amp/check_finite_and_unscale_op_xpu.cc [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+                              is_finite.data<bool>(), x->numel());
+        PADDLE_ENFORCE_EQ(r, XPU_SUCCESS, platform::errors::External(
+                                              "XPU API(isfinite) return wrong "
+                                              "value[%d %s]",
+                                              r, XPUAPIErrorMsg[r]));
+        r = xpu::logical_not(dev_ctx.x_context(), reinterpret_cast<const bool*>(
+                                                      is_finite.data<bool>()),
+                             is_finite.data<bool>(), x->numel());
+        PADDLE_ENFORCE_EQ(r, XPU_SUCCESS, platform::errors::External(
+                                              "XPU API(isfinite) return wrong "
```
Comment (QingshuChen) [inline paddle/fluid/operators/amp/check_finite_and_unscale_op_xpu.cc]: logical_not

File: paddle/fluid/operators/amp/check_finite_and_unscale_op_xpu.cc [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+                       is_nan.data<bool>(), x->numel());
+        PADDLE_ENFORCE_EQ(r, XPU_SUCCESS, platform::errors::External(
+                                              "XPU API(isnan) return wrong "
+                                              "value[%d %s]",
+                                              r, XPUAPIErrorMsg[r]));
+        r = xpu::logical_or(dev_ctx.x_context(), is_finite.data<bool>(),
+                            is_nan.data<bool>(), is_finite.data<bool>(),
+                            x->numel());
+        PADDLE_ENFORCE_EQ(r, XPU_SUCCESS, platform::errors::External(
+                                              "XPU API(any) return wrong "
```
Comment (QingshuChen) [inline paddle/fluid/operators/amp/check_finite_and_unscale_op_xpu.cc]: logical_or

File: paddle/fluid/operators/amp/check_finite_and_unscale_op_xpu.cc [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+                                              r, XPUAPIErrorMsg[r]));
+
+        r = xpu::cast_v2(dev_ctx.x_context(), float_out.data<MPDType>(),
+                         reinterpret_cast<float16*>(out->data<T>()),
+                         out->numel());
+
+        PADDLE_ENFORCE_EQ(r, XPU_SUCCESS, platform::errors::External(
+                                              "XPU API(cast_v2) return wrong "
+                                              "value[%d %s]",
+                                              r, XPUAPIErrorMsg[r]));
```
Comment (QingshuChen) [inline paddle/fluid/operators/amp/check_finite_and_unscale_op_xpu.cc:139]: 加xpu_wait

File: paddle/fluid/operators/amp/update_loss_scaling_op_xpu.cc [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+      T* out_data = out->mutable_data<T>(dev_ctx.GetPlace());
+      int num = out->numel();
+      if (cpu_found_inf_data) {
+        VLOG(1) << "-- UpdateLossScaling: Find infinite grads. --";
+        int r = 0;
+        r = xpu::constant(dev_ctx.x_context(),
+                          reinterpret_cast<XPUTyp*>(out_data), num,
+                          XPUTyp(0.0));
+        PADDLE_ENFORCE_EQ(r == xpu::Error_t::SUCCESS, true,
+                          platform::errors::External(
```
Comment (QingshuChen) [inline paddle/fluid/operators/amp/update_loss_scaling_op_xpu.cc]: 这个错误提示改下，参考check_finite_and_unscale_op_xpu.cc

File: paddle/fluid/operators/dropout_op_xpu.cc [hunk 5]
```diff
@@ -127,9 +104,33 @@ class DropoutGradXPUKernel : public framework::OpKernel<T> {
...
+      float scale =
+          (dropout_prob == 1.0f) ? (1.0f) : (1.0f / (1.0f - dropout_prob));
+      int r = xpu::scale(dev_ctx.x_context(),
+                         reinterpret_cast<const XPUTyp*>(mask->data<T>()),
+                         reinterpret_cast<XPUTyp*>(mask_new.data<T>()),
+                         mask->numel(), false, scale, 0.0f);
+      PADDLE_ENFORCE_EQ(
+          r, xpu::Error_t::SUCCESS,
+          platform::errors::External(
+              "XPU dropout return wrong value[%d], please check whether "
```
Comment (QingshuChen) [inline paddle/fluid/operators/dropout_op_xpu.cc]: 错误提示改一下

**Review by QingshuChen — CHANGES_REQUESTED** [review]

**Review by QingshuChen — APPROVED** [review]
LGTM

**Review by jiweibo — APPROVED** [review]
LGTM for xpu.cmake

**Pull request merged**